sha2/src/sha256-avx2-asm.S

```diff
@@ -0,0 +1,802 @@
+# Copied and slightly modified
+# from the linux kernel arch/x86/crypto/sha256-avx2-asm.S
+# by Harald Hoyer <harald@hoyer.xyz>
+
+########################################################################
+# Implement fast SHA-256 with AVX2 instructions. (x86_64)
+#
+# Copyright (C) 2013 Intel Corporation.
+#
+# Authors:
+#     James Guilford <james.guilford@intel.com>
+#     Kirk Yap <kirk.s.yap@intel.com>
+#     Tim Chen <tim.c.chen@linux.intel.com>
+#
+# This software is available to you under a choice of one of two
+# licenses.  You may choose to be licensed under the terms of the GNU
+# General Public License (GPL) Version 2, available from the file
+# COPYING in the main directory of this source tree, or the
+# OpenIB.org BSD license below:
```

Comment:
Licensing-wise all of our crates are MIT+Apache 2.0 which is the conventional license choice for Rust projects

Reply:
I know.. this is why I was asking. It was just easy to copy from the linux kernel.

Reply:
This would be an alternative https://github.com/openssl/openssl/blob/13a574d8bb2523181f8150de49bc041c9841f59d/crypto/sha/asm/sha256-mb-x86_64.pl with `Apache License 2.0` and even faster in my benchmarks.

Reply:
That might work. We could potentially contact the author to ask about dual licensing under MIT.

Reply:
@tarcieri The version from openssl (in latest git) is effectively triple-licensed under 3-clause BSD, Apache 2.0, and GPL, I think the combination of 3-clause BSD and Apache 2.0 is sufficiently equivalent that it'd be fine to just document that as the license rather than requesting what amounts to a relicense from BSD to MIT (both permissive licenses that are nearly equivalent).

Reply:
@joshtriplett per @newpavlov's [comment below](https://github.com/RustCrypto/asm-hashes/pull/32#issuecomment-802038199), we can just change the license of the crate to be whatever license the original ASM uses.